syntax = "proto3";

option java_multiple_files = true;
option java_package = "singleTerminaLoginSystem.protos.loginManageService";
option java_outer_classname = "loginManageProto";
option objc_class_prefix = "TXH";

package proto.userLoginManage;


service userLoginManageService {
  // regist
  rpc regist (stream RegistRequest) returns (stream RegistResponse) {}

  // login
  rpc login (stream LoginRequest) returns (stream LoginResponse) {}

  // logout
  rpc logout (stream LogoutRequest) returns (stream LogoutResponse) {}

  // heart beat
  rpc heartBeat (stream HeartBeatRequest) returns (stream HeartBeatResponse) {}

  // kick out user
  rpc kickOutUser (stream KickOutUserRequest) returns (stream KickOutUserResponse) {}
  rpc queryLoginStatus (stream QueryLoginStatusRequest) returns (stream QueryLoginStatusResponse) {}
}


//***
//@request  BasicRequest
//@response BasicResponse
//@remark   请求的基本信息
//***
message BasicRequest
{
	int64   timestamp     = 1; //发送消息的当前时间戳
	string  uuid          = 2; //客户端的uuid
}

message BasicResponse
{
	int64   timestamp     = 1; //时间戳
	int32   code          = 2; //状态码
	string  msg           = 3; //描述信息
}


//***
//@request  RegistRequest
//@response RegistResponse
//@remark   用户注册
//***
message RegistRequest
{
	BasicRequest             basic     = 1;
	string                   user_name = 2;
	string                   pass_word = 3;
}

message RegistResponse
{
	BasicResponse            basic     = 1;
}


//***
//@request  LoginRequest
//@response LoginResponse
//@remark   用户登录
//***
message LoginRequest
{
	BasicRequest             basic     = 1;
	string                   user_name = 2;
	string                   pass_word = 3;
}

message LoginResponse
{
	BasicResponse            basic     = 1;
}


//***
//@request  LogoutRequest
//@response LogoutResponse
//@remark   退出登录
//***
message LogoutRequest
{
	BasicRequest             basic     = 1;
	string                   user_name = 2;
}

message LogoutResponse
{
	BasicResponse            basic     = 1;
}

//***
//@request  KickOutUserRequest
//@response KickOutUserResponse
//@remark   踢出用户
//***
message KickOutUserRequest
{
	BasicRequest             basic     = 1;
	string                   user_name = 2;
}

message KickOutUserResponse
{
	BasicResponse basic                = 1;
}


//***
//@request  QueryLoginStatusRequest
//@response QueryLoginStatusResponse
//@remark   询问登录状态
//***
message QueryLoginStatusRequest
{
	BasicRequest             basic     = 1;
}

message QueryLoginStatusResponse
{
	BasicResponse            basic     = 1;
	int32                    operation = 2; //应处理的动作 1.无须处理  2. 有新设备登录，此设备需退出登录
}

